galsim/config/input_powerspectrum.py

```diff
@@ -55,6 +84,23 @@ def PowerSpectrumInit(ps, config, base):
     ps.buildGrid(grid_spacing=grid_spacing, ngrid=ngrid, center=world_center,
                  rng=base['rng'], interpolant=interpolant)
 
+    # Make sure this process gives consistent results regardless of the number of processes
+    # being used.
+    if not isinstance(ps, galsim.PowerSpectrum):
+        # Then ps is really a proxy, which means the rng was pickled, so we need to
+        # discard the same number of random calls from the one in the config dict.
```

Comment:
It seems like there out to be a more direct way to determine whether an object is a proxy.  If it's something you have control over (I haven't followed the logic carefully enough to know if it's something from the built-in `multiprocessing` module), maybe an `is_proxy` class attribute?